File: tests/e2e/fixtures/mock_repo/hooks/e2e_test_hook.py
#!/usr/bin/env python3
"""E2E test hook script for validation.

This hook is triggered by PostToolUse events for Edit/MultiEdit/Write operations.
It receives a config file path as the first argument.
"""

import json
import sys
from pathlib import Path


def main() -> int:
    """Process hook invocation and validate input."""
    # Read config if provided
    config_path = Path(sys.argv[1]) if len(sys.argv) > 1 else None

    # Read stdin (Claude Code passes event data via stdin)
    input_data = sys.stdin.read()

    try:
        event = json.loads(input_data) if input_data.strip() else {}
    except json.JSONDecodeError:
        event = {}

    # Simple validation: just pass through
    # In real hooks, you would process the event and return modified data
    config_loaded = False
    if config_path is not None and config_path.exists():
        config_loaded = True

    result = {
        'continue': True,
        'message': 'E2E test hook executed successfully',
        'config_loaded': config_loaded,
        'event_type': event.get('type', 'unknown'),
    }

    print(json.dumps(result))
    return 0


if __name__ == '__main__':
    sys.exit(main())

File: tests/e2e/fixtures/mock_repo/hooks/e2e_test_hook.js
#!/usr/bin/env node
/**
 * E2E test JavaScript hook script for validation.
 *
 * This hook is triggered by PostToolUse events for Read operations.
 * Validates that JavaScript hooks receive proper node prefix.
 */

'use strict';

function main() {
    // Read stdin (Claude Code passes event data via stdin)
    let inputData = '';

    process.stdin.setEncoding('utf8');
    process.stdin.on('readable', () => {
        let chunk;
        while ((chunk = process.stdin.read()) !== null) {
            inputData += chunk;
        }
    });

    process.stdin.on('end', () => {
        let event = {};
        try {
            event = inputData.trim() ? JSON.parse(inputData) : {};
        } catch (e) {
            event = {};
        }

        const result = {
            continue: true,
            message: 'E2E JavaScript hook (.js) executed successfully',
            runtime: 'node',
            event_type: event.type || 'unknown',
        };

        console.log(JSON.stringify(result));
    });
}

main();
